media/server/main/source/MediaKeysServerInternal.cpp

```diff
@@ -691,4 +691,49 @@ void MediaKeysServerInternal::ping(std::unique_ptr<IHeartbeatHandler> &&heartbea
 
     m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
 }
+
+MediaKeyErrorStatus MediaKeysServerInternal::getMetricSystemData(std::vector<uint8_t> &buffer)
+{
+    RIALTO_SERVER_LOG_DEBUG("entry:");
+
+    uint32_t bufferLength{1024};
+    const uint32_t kMaxBufferLength{65536};
+    MediaKeyErrorStatus status;
+    buffer.resize(bufferLength);
+
+    for (int attempts = 0; bufferLength <= kMaxBufferLength; ++attempts)
+    {
+        auto task = [&]() { status = m_ocdmSystem->getMetricSystemData(bufferLength, buffer); };
+        m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
+
+        if (status != MediaKeyErrorStatus::BUFFER_TOO_SMALL)
+        {
+            RIALTO_SERVER_LOG_ERROR("Buffer too small with a size of %u", bufferLength);
```

Comment:
1. It's actually the opposite
2. This log shouldn't be an error, because its also called when we get OK response
3. This log is actually not needed at all, because error code will be logged after for loop